### ISSUE TITLE
Scratch.vim doesn't exit cleanly on winleave

### ISSUE THREAD

**Issue by mtglsk** [mtth/scratch.vim#12]
I have some autocmd's on winenter and winleave, which afaik don't work with your plugin.

`autocmd WinLeave * set cul`
`autocmd WinEnter * set nocul`

**Comment by mtth** [comment]
Indeed, I can reproduce this. I'll think how best to can change the way the auto-closing is handled.

If you're curious I believe this happens because by default `autocmd`s don't trigger recursively (more details here - [`autocmd-nested`](http://vimdoc.sourceforge.net/htmldoc/autocmd.html#autocmd-nested)).

**Comment by mtth** [comment]
Fixed in 187d1d5fe4437e6af908c18b94e06ffa9414c6a5. Let me know if you run into any issues.

Note that for your `autocmd`s to work reliably (e.g. also with `split`) you should change them to:
```viml
autocmd VimEnter,WinEnter,BufWinEnter * set cul
autocmd WinLeave * set nocul
```

**Comment by mtglsk** [comment]
Works great! Very much thanks!
